frontend/public/components/marketplace/kubernetes-marketplace.jsx

```diff
@@ -0,0 +1,132 @@
+import * as React from 'react';
+import * as _ from 'lodash-es';
+import * as PropTypes from 'prop-types';
+import { Helmet } from 'react-helmet';
+
+import {Firehose, PageHeading, StatusBox} from '../utils';
+import {referenceForModel} from '../../module/k8s';
+import {normalizeIconClass} from '../catalog-item-icon';
+import {PackageManifestModel} from '../../models';
+import CatalogTileView from 'patternfly-react-extensions/dist/esm/components/CatalogTileView/CatalogTileView';
+import CatalogTile from 'patternfly-react-extensions/dist/esm/components/CatalogTile/CatalogTile';
+
+const normalizePackageManifests = (packageManifests, kind) => {
+  const activePackageManifests = _.filter(packageManifests, packageManifest => {
+    return !packageManifest.status.removedFromBrokerCatalog;
+  });
+  return _.map(activePackageManifests, packageManifest => {
+    const tileName = packageManifest.metadata.name;
+    const iconClass = 'fa fa-clone'; // TODO: get this info from the packagemanifest
+    const iconObj = _.get(packageManifest, 'status.channels[0].currentCSVDesc.icon[0]', null);
+    const tileImgUrl = iconObj ? `data:${_.get(iconObj, 'mediatype')};base64,${_.get(iconObj, 'base64data')}` : null;
+    const tileIconClass = tileImgUrl ? null : iconClass;
+    const tileDescription = packageManifest.metadata.description;
+    const tileProvider = packageManifest.metadata.labels.provider;
+    const tags = packageManifest.metadata.tags;
+    return {
+      obj: packageManifest,
+      kind,
+      tileName,
+      tileIconClass,
+      tileImgUrl,
+      tileDescription,
+      tileProvider,
+      tags,
+    };
+  });
+};
+
+const getItems = (props) => {
+  const {packagemanifests, loaded} = props;
+  let packageManifestItems = null;
+  if (!loaded) {
+    return [];
+  }
+  if (packagemanifests) {
+    packageManifestItems = normalizePackageManifests(packagemanifests.data, 'PackageManifest');
+  }
+  return _.sortBy([...packageManifestItems], 'tileName');
+};
+
+class MarketplaceListPage extends React.Component {
+  state = {};
+
+  static getDerivedStateFromProps(props, state) {
+    const {packagemanifests} = props;
+    if (packagemanifests !== _.get(state, 'packagemanifests')) {
+      const items = getItems(props);
+      return {items, packagemanifests};
```

Comment:
It feel like an anti pattern to have both props.packagemanifests and state.packagemanifests, although I don't have a great suggestion except maybe to use a different lifecycle method.

I know this was discussed a lot earlier :)

Reply:
Probably better to keep the `packagemanifests.data` in the state and check if that changed rather than the entire result object anyhow. Could normalize first and check items but that would be extra, often unnecessary, processing.